pyomo/gdp/plugins/hull.py

```diff
@@ -924,25 +758,27 @@ def _transform_constraint(self, obj, disjunct, var_substitute_map,
                     newConsExpr = expr - (1-y)*h_0 <= c.upper*y
 
                 if obj.is_indexed():
-                    newConstraint.add((i, 'ub'), newConsExpr)
+                    newConstraint.add((name, i, 'ub'), newConsExpr)
                     # map (have to account for fact we might have created list
                     # above
                     transformed = constraintMap['transformedConstraints'].get(c)
                     if transformed is not None:
-                        transformed.append(newConstraint[i,'ub'])
+                        transformed.append(newConstraint[name, i, 'ub'])
```

Comment:
Codecov is crying that this line is not covered by any test, but I think it's alright.